File: tabulation/javascript/gridTraveler.js
function gridTraveler(m, n) {
    const table = Array(m + 1).fill(0).map(() => Array(n + 1).fill(0))
    table[1][1] = 1
    
    for (let i = 0; i <= m; i++) {
        for (let j = 0; j <= n; j++) {
            if (j !== n) table[i][j + 1] += table[i][j]
            if (i !== m) table[i + 1][j] += table[i][j]
        }
    }

    return table[m][n]
}

console.log(gridTraveler(8, 18))
console.log(gridTraveler(4, 4))
console.log(gridTraveler(3, 3))
console.log(gridTraveler(18, 18))

File: tabulation/kotlin/gridTraveler.kt
fun gridTraveler(m: Int, n: Int): Int {
    val table: Array<Array<Int>> = Array(m + 1) { Array(n + 1) { 0 } }
    table[1][1] = 1

    for (i: Int in 0..m) {
        for (j: Int in 0..n) {
            if (j != n) table[i][j + 1] += table[i][j]
            if (i != m) table[i + 1][j] += table[i][j]
        }
    }

    return table[m][n]
}

fun main(vararg args: String) {
    when (args.size) {
        2 -> println(gridTraveler(args[0].toInt(), args[1].toInt()))
        else -> println(gridTraveler(3, 3))
    }
    
}